server/modules/hdVizManager/HdVizManager.js

```
@@ -0,0 +1,139 @@
+ *  of the Dataset, DistanceMap and the graphs once they were created.
+ *
+ *  @param {Dataset | null} dataset One only for application by design.
+ *  @param {Array<GraphModel>} graphs All the different instances of graph that were created.
+ *  @param {Integer | Null} current The current graph that is visualized
+ *
+ * */
+export default function HdVizManager(dataset, graphs= [], current = -1){
+
+    /* Inheritance from Subject*/
+    Subject.call(this)
+
+    privates.set(this, {dataset: dataset, graphs: [], current: current})
+    filterGraphs.call(this, graphs)
+}
+
+/** Subject inheritance */
+HdVizManager.prototype = Object.create(Subject.prototype)
+HdVizManager.prototype.constructor = HdVizManager
+
+HdVizManager.prototype.setDataset = function (loadedDataset){
+
+    const _this = privates.get(this)
+
+    /** Make new dataset assignment. (type check?) */
+    _this.dataset = loadedDataset
+
+    /** Notify change of dataset. */
+    this.notify(new Event(EventType.DATA_LOADED, null, this))
+
+    /* Destroy all graphs */
+    _this.graphs = []
+    _this.current = - 1
+
+}
+
+/** @description Add a graph to the array of currently graphModels created with the actual dataset. It is automatically
+ *              set as the current graph. Items from the same class are not allowed. (We could overwrite ?)
+ *
+ *  @param {GraphModel} graphModel Model to add to the list of graphs. It is automatically se t as the current.
+ *  @return {Boolean} Returns true if the graph was possible to add (no repetition of type), else false.
+ * */
+HdVizManager.prototype.addGraph = function (graphModel){
+
+    const _this = privates.get(this)
+
+    /* We have to avoid duplicates. */
+    const matches = _this.graphs.filter(graph => graph instanceof graphModel.constructor)
+
+    /** If we have a match we can't add the graph to our models list, it would be stupid.
+     *  Do we want to throw exception or simply return false and do nothing? */
```

Direi che sia meglio lanciare eccezioni e lasciare al chiamante gestirle ("fail fast" altrimenti)